code/hud/hudtarget.cpp

```diff
@@ -5776,9 +5779,10 @@ void HudGaugeWeaponEnergy::render(float  /*frametime*/)
 			}
 		}
 
-		for ( i = 0; i < sw->num_primary_banks; i++ )
+		// maybe flash the energy bar
+		for ( i = 0; i < sw->num_primary_banks + sw->num_secondary_banks; i++ )
```

Comment:
Is this addition of the extra bank capacity going to mess with the indexing at all?

Also keep in mind that the bank offsets may be different.  If a ship has 1 primary and 2 secondary weapons, it will have the same total number as 2 primary and 1 secondary, but the indexes will be different.

Reply:
Yes, this is intentional, these *aren't* being used to index into weapons anymore, these are essentially changed into indices for the weapon lines in the weapon gauge itself. Whether it has 1 primary/2 secondary or 2 primary/1 secondary, index one is always going to the be the second line displayed on the hud and is therefore index 1, regardless. It only checked up to the num primary banks because it used to only flash the energy bar if any primary weapon line was flashing. Now it checks all of the currently used weapons, and only cares if that one in particular asked for a flashing energy bar, regardless whether it's a primary or secondary.

Reply:
Gotcha.